sklearn/cluster/_kmeans.py

```diff
@@ -1287,6 +1299,17 @@ def __init__(
         self.algorithm = algorithm
 
     def _check_params(self, X):
+        # TODO(1.3): Remove
+        if self.n_init == "warn":
+            warnings.warn(
+                "The default value of `n_init` will change from 10 to 'auto' in 1.3."
+                " Set the value of `n_init` explicitly to suppress the warning",
+                FutureWarning,
+            )
+            self.n_init = "auto"
+        if self.n_init == "auto":
+            self.n_init = 1 if self.init == "kmeans++" else 10
```

Comment:
We can not change `__init__` parameters in `fit`.

```suggestion
            self._n_init = 1 if self.init == "kmeans++" else 10
```

The rest of the code need to use `self._n_init`.